Sample Applications/CustomComboBox/CustomComboBox.csproj

```diff
@@ -1,101 +1,14 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+﻿<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{9E3EECAF-86CF-45CB-8ABB-01A850A308F0}</ProjectGuid>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+    <UseWPF>true</UseWPF>
     <OutputType>WinExe</OutputType>
     <RootNamespace>CustomComboBox</RootNamespace>
     <AssemblyName>CustomComboBox</AssemblyName>
-    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
-    <ProjectTypeGuids>{60dc8134-eba5-43b8-bcc9-bb4bc16c2548};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
-    <WarningLevel>4</WarningLevel>
-    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
```

Comment:
I got CS0579 errors - Duplicate 'System.Rdflection.AssemblyCompanyAttribute' attribute. This was not restricted to just AssemblyCompanyAttribute. There were other attributes as well: AssemblyConfigurationAttribute, File version attribute, product attribute etc. By setting GenerateAssemblyInfo to false, these errors went away

Reply:
That makes sense. That's happening because of this file: 

https://github.com/microsoft/WPF-Samples/blob/4e7a746cbe69df943efa0d57ef45ac3325ea3a49/Sample%20Applications/CustomComboBox/Properties/AssemblyInfo.cs#L7-L17

I didn't see `GenerateAssemblyInfo=false` in all other projects, so I digged in a bit. See for e.g. [`DropShadow.csproj`](https://github.com/microsoft/WPF-Samples/blob/4e7a746cbe69df943efa0d57ef45ac3325ea3a49/Sample%20Applications/DropShadow/DropShadow.csproj). 

- The project has [Properties\AssemblyInfo.cs](https://github.com/microsoft/WPF-Samples/blob/4e7a746cbe69df943efa0d57ef45ac3325ea3a49/Sample%20Applications/DropShadow/Properties/AssemblyInfo.cs)
- It does not specify `GenerateAssemblyInfo=false`
- I would expect it to spew the same errors you saw in `CustomComboBox`. 

The way it avoids it is like this: 

https://github.com/microsoft/WPF-Samples/blob/4e7a746cbe69df943efa0d57ef45ac3325ea3a49/Sample%20Applications/DropShadow/DropShadow.csproj#L5

This effectively ensures that only .cs files explicitly included in `Compile` items are included. In other words, it is no longer `**\*.cs`. 

Then it specifies only the .cs files, leaving out 'Properties\AssemblyInfo.cs`: 

https://github.com/microsoft/WPF-Samples/blob/4e7a746cbe69df943efa0d57ef45ac3325ea3a49/Sample%20Applications/DropShadow/DropShadow.csproj#L49-L68

This way it doesn't use the `AssemblyInfo.cs` file, and lets the compiler auto-generate these `[assembly:]` attributes. This means that we have vestigial files that need to be deleted. 

Auto-generation of these attributes is the preferred approach in .NET Core.

Reply:
This helped in understanding it better. Thanks Vatsan.